File: jiwon/programmers/cs/run_race.cs
    using System;
    using System.Collections.Generic;

    public class Solution {
        public string[] solution(string[] players, string[] callings) {
            Dictionary<string, int> p = new Dictionary<string, int>();


            for (int i = 0; i < players.Length; i++) {
                p[players[i]] = i;
            }

            foreach (var call in callings) {
                int index = p[call];
                p[call] -= 1;
                p[players[index - 1]] += 1;
                string temp = players[index];
                players[index] = players[index - 1];
                players[index - 1] = temp;

            }
            return players;
        }
    }

File: jiwon/programmers/c++/run_race.cpp
#include <string>
#include <map>
#include <vector>

using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    map<string, int> race;
    
    for (int i = 0; i < players.size(); i++){
        race[players[i]] = i;
    }
    
    for (const auto& c : callings) {
        int i = race[c];
        race[c] -= 1;
        string temp = players[i - 1];
        race[temp] += 1;
        players[i - 1] = players[i];
        players[i] = temp;
    }
    
    return players;
}